transport/http2_client.go

```diff
@@ -110,15 +109,15 @@ func newHTTP2Client(addr string, authOpts []credentials.Credentials) (_ ClientTr
 			// multiple ones provided. Revisit this if it is not appropriate. Probably
 			// place the ClientTransport construction into a separate function to make
 			// things clear.
-			conn, connErr = ccreds.Dial(addr)
+			conn, connErr = ccreds.DialTimeout(addr, timeout)
```

Comment:
Shouldn't matter. If the user hasn't specified a timeout, it'll just be 0, which is the default for a new net.Dialer anyway.